src/main/java/characterpicker/infrastructure/CharacterParser.java

```diff
@@ -3,26 +3,55 @@
 import characterpicker.domain.Character;
 import org.apache.commons.lang.StringUtils;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
 public class CharacterParser {
-    private CharacterParser(){
+    private CharacterParser() {
         throw new IllegalStateException("Utility class");
     }
 
     static Character parseCharacter(String unparsedCharacter) {
 
-        String[] characterProperties = unparsedCharacter.split(",");
+        if (unparsedCharacter.endsWith(";")){
+            throw new IllegalArgumentException("Completion required. If you don't have one type 'none'");
+        }
+
+        String[] unparsedProperties = unparsedCharacter.split(";");
+        String character = unparsedProperties[0];
+        String completion = unparsedProperties[1];
+
+        String[] characterProperties = character.split(",");
         if (characterProperties.length < 2) {
             throw new IllegalArgumentException("Name and weight required");
         }
         String name = characterProperties[0];
         String unparsedWeight = characterProperties[1];
+
         if (!StringUtils.isNumeric(unparsedWeight)) {
             throw new IllegalArgumentException("Provided weight is not a number");
         }
 
-
-        int weight = Integer.parseInt(unparsedWeight);
+        int weight = Integer.parseInt(unparsedWeight) + parseCompletion(completion);
```

Comment:
Tu możesz dostać overflowa, jak ktoś poda bardzo wysoki wieght `2147483647` + parę makrów i wyjdzie liczba ujemna, Też by się przydał na to test.

https://www.baeldung.com/java-overflow-underflow#2-throw-an-exception